runtime/vm/jvminit.c

```diff
@@ -5175,6 +5175,50 @@ static void testOptionValueOps(J9JavaVM* vm) {
 	TEST_INT(uResult, (1024*1024*1024));
 	TEST_INT(intResult, OPTION_OK);
 
+#if defined(J9VM_ENV_DATA64)
+	SET_TO(1, "-Xfog8388607T");			/* 2^63 */
+	optName = "-Xfog";
+	intResult = GET_MEMORY_VALUE(1, optName, uResult);
+	TEST_INT(uResult, (8388607*1024L*1024L*1024L*1024L));
+	TEST_INT(intResult, OPTION_OK);
+
+	SET_TO(1, "-Xfoh1t");
+	optName = "-Xfoh";
+	intResult = GET_MEMORY_VALUE(1, optName, uResult);
+	TEST_INT(uResult, (1024*1024*1024*1024L));
+	TEST_INT(intResult, OPTION_OK);
+
+	SET_TO(1, "-Xfoi8388608T");			/* 2^63+1 */
+	optName = "-Xfoi";
+	intResult = GET_MEMORY_VALUE(1, optName, uResult);
+	TEST_INT(intResult, OPTION_OVERFLOW);
+
+	SET_TO(1, "-Xfoj8388608t");			/* 2^63+1 */
+	optName = "-Xfoj";
+	intResult = GET_MEMORY_VALUE(1, optName, uResult);
+	TEST_INT(intResult, OPTION_OVERFLOW);
+#else	/* !defined(J9VM_ENV_DATA64) */
```

Comment:
This comment should also be fixed.

Reply:
@pshipton , alright, fixed the comment in `jvminit.c`.

I have some questions:
1.  The size suffix parsing logic seems to be duplicated in `vmargs.c` and in `mmparse.cpp`, as pointed out by @dmitripivkine.
Is there a reason for such duplication?
Which one is used for which case?
2. When I un-commented the `#define JVMINIT_UNIT_TEST` in `jvminit`.c, I got some failed tests, even before changing it.
When I added the unit tests for overflow cases, I see I added more failed tests, in spite of my effort to detect that the overflow error was expected.
I would love to get more insights on this area, because I am feeling uncomfortable to get a test summary result being "Failed", even before my changes.
Maybe should I open another incident to track this unit testing issue, as it's kind of independent of current issue?

Reply:
Yes, you can open an Issue for any failures that occur previous to your change.

The `mmparse.cpp` code is used to parse options that affect the object heap sizes, like `-Xmx` or `-XX:MaxHeapSize`. The reason for having duplicate code is historic, as the code bases used to be separated.

Reply:
@keithc-ca I have included all your remarks in my latest commit. Please let me know if there is anything else that's missing to get this PR integrated.